src/Util/Asserts.h

```diff
@@ -13,24 +13,40 @@
 #define CONVERT_TO_STRING(X) #X
 #define CONVERT_EXPR_TO_STRING(X) CONVERT_TO_STRING(X)
 
-#define OWSAssert(X) \
-if (!(X)) { \
-NSLog(@"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-}
-
-// OWSAssert() should be used in Obj-C and Swift methods.
-// OWSCAssert() should be used in free functions.
-#define OWSCAssert(X) \
-if (!(X)) { \
-NSLog(@"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-NSCAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-}
+// OWSAssert() and OWSFail() should be used in Obj-C methods.
+// OWSCAssert() and OWSCFail() should be used in free functions.
+
+#define OWSAssert(X)                                                                                                   \
+    if (!(X)) {                                                                                                        \
+        NSLog(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                             \
+        NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                               \
+    }
+
+#define OWSCAssert(X)                                                                                                  \
+    if (!(X)) {                                                                                                        \
+        NSLog(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                             \
+        NSCAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                              \
+    }
+
+#define OWSFail(X)                                                                                                     \
```

Comment:
Awesome. Thanks!

On a more radical note, what if we defined these in non-debug, and changed the NSLog to DDLogError so that we can see them in production logs (and eventually they could be reported).

No need to do that in this PR, if ever.